pkg/templates/generic_lookup_func.go

```diff
@@ -297,3 +299,103 @@ func onAllowlist(allowlist []ClusterScopedObjectIdentifier, rsrc ClusterScopedOb
 
 	return false
 }
+
+func (t *TemplateResolver) getNodesWithExactRolesHelper(
+	options *ResolveOptions,
+	templateResult *TemplateResult,
+) func(...string) (
+	map[string]interface{}, error,
+) {
+	return func(name ...string) (
+		map[string]interface{}, error,
+	) {
+		return t.getNodesWithExactRoles(options, templateResult, name...)
+	}
+}
+
+// function getNodesWithExactRoles gets a listing of all nodes with only the
+// role specified.  Any nodes that include other roles in addition
+// to the specified roles are not included.
+func (t *TemplateResolver) getNodesWithExactRoles(
+	options *ResolveOptions,
+	templateResult *TemplateResult,
+	name ...string,
+) (
+	map[string]interface{}, error,
+) {
+	result := []unstructured.Unstructured{}
+
+	if len(name) == 0 || strings.TrimSpace(name[0]) == "" {
```

Comment:
Why is this only checking if the first role is not empty? Should this be in a for loop to check all of them?

Reply:
Corrected to check if the value specified is empty before adding it to the list, then error check if the search list is empty